lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js

```diff
@@ -194,7 +193,7 @@ function handleLogs() {
     const destinationArn = {
       op: 'replace',
       path: '/accessLogSettings/destinationArn',
-      value: `arn:aws:logs:${region}:${this.accountId}:log-group:${logGroupName}`,
+      value: provider.logs.restApi.iamRoleArn || `arn:aws:logs:${region}:${this.accountId}:log-group:${logGroupName}`,
```

Comment:
This property sets `logGroupName` not IAM role.

The IAM role is currently set, through this custom resource -> https://github.com/serverless/serverless/blob/b2091b030ed788af5f8b3e3345ce878c651faae4/lib/plugins/aws/package/compile/events/apiGateway/lib/stage/index.js#L92-L100

So when existing IAM role is provided, we should not configure this custom resource, but just configure APIGW account's CloudWatch role setting to its value, static way (like it was in done prior 1.50 -> https://github.com/serverless/serverless/blob/v1.49.0/lib/plugins/aws/package/compile/events/apiGateway/lib/stage.js#L155-L160 )